lib/explorer/data_frame.ex

```diff
@@ -516,7 +516,7 @@ defmodule Explorer.DataFrame do
 
     * `:max_rows` - Maximum number of lines to read. (default: `nil`)
 
-    * `:null_character` - The string that should be interpreted as a nil value. (default: `"NA"`)
+    * `:nil_values` - A string or a list of strings that should be interpreted as a nil values. (default: `nil`)
```

Comment:
Given it is a list, do we need to support `nil`? `nil` and empty list should be the same.  So my suggestion would be to require this to be a list. WDYT?

Reply:
Also, are empty entries always `nil`? Can we turn that off?

Reply:
>  So my suggestion would be to require this to be a list.

I agree! The API should accept only string or a list of strings - and the default is an empty list -, but I think we can pass down as an Option to the NIF, so it's slightly more efficient there. WDYT?

Reply:
Nah, I don't think the empty list vs an option makes a large difference in terms of efficiency. I would only allow lists (not even just strings) and nothing more. :) Especially as the name is `nil_values` (plural).

Reply:
8349ff32ac257ce8ebd2e7d431d9e16f388ea01f.

It looks like there's maybe some room for more consistency here. I noticed that, for example, `dtypes` can't be `nil` but to the NIF they're `Option` and `columns` can be `list` | `nil`.

Usually I like to scan for high-impact contribs, but if this is something you'd like a PR for I'm happy to send it as well. I can probably be more useful this way while I learn Elixir :)

Reply:
@cnpryer for columns it makes sense to be list | nil because an empty list would be no columns (even though it makes no sense, at least semantically it has a different meaning). I think though we can remove the Option from dtypes. :) PRs are always welcome!